Improve state explanation (#95)
iov-one/docs: docs/weave-tutorial/02-domain.md

> The first thing we consider is the data we want to store, which is known as the state. After that, we can focus on the messages, which trigger state transitions. All blockchain state must be stored in our Merkle-ized store, which provides validity hashes and proofs. This is exposed to the application as a basic key-value store, which also allows in-order iteration over the keys. On top of this, we have built some tools, such as secondary indexes and sequences, similar to how [Storm adds an ORM](https://github.com/asdine/storm#simple-crud-system) on top of [BoltDB’s key-value store](https://github.com/boltdb/bolt#using-buckets). We have avoided struct tags and tried to type as strictly as we can (without using generics).

Reviewer: ```suggestion
The first thing we consider is the data we want to store, which is known as the state. After that, we can focus on the messages, which trigger state transitions. All blockchain state must be stored in our Merkle-ized store, which provides validity hashes and proofs. This is exposed to the application as a basic key-value store, which also allows in-order iteration over the keys. On top of this we have built some tools, such as secondary indexes and sequences, similar to how [Storm adds an ORM](https://github.com/asdine/storm#simple-crud-system) on top of [BoltDB’s key-value store](https://github.com/boltdb/bolt#using-buckets). We have avoided struct tags and tried to type as strictly as we can (without using generics).
```

## State

> State as in [Turing machine](https://en.wikipedia.org/wiki/Turing_machine), is how the system's data is presented this contained. We are building an Order book application to solidify the state concept. Exchange topic is quite hot at the time of writing this tutorial.

Reviewer: I don't understand hw this first sentence ends: 'this contained' 

Looks like a typo?

> - **Markets** will contain rules as to who (which public keys) may create an **order book** or who can cancel an order. The term _market_ here, defines the state of an entity that represents the real markets. The blockchain application may have multiple markets, and each market may have multiple Orderbooks; however, each token pair can only have one order book per market.

Reviewer: ```suggestion
- **Markets** will contain rules as to who (which public keys) may create an **order book** or who can cancel an order. The term _market_ here defines the state of an entity that represents the real markets. The blockchain application may have multiple markets, and each market may have multiple Orderbooks; however, each token pair can only have one order book per market.
```

> There is no global chain owner, but each market has its own chain owner that adds Order books and sets fees. This could be one person, a multisig, or a governance contract (DAO). When adding an order book, the market with a given market ID is checked, then looks up the owner _for that market_. Rather than having one owner that can create Order books for all markets, each market stores the information on which owner can update it.

Reviewer: What does DAO stand for?

> - **Order book** state will defines which Ask/Bid market it holds and the number of available orders. Order book will contain orders and trades.

Reviewer: ```suggestion
- **Order book** state will define which Ask/Bid market it holds and the number of available orders. An Order book will contain orders and trades.
```
